src/base/registrations/registrar.ts

```diff
@@ -99,44 +111,40 @@ export async function getRegistration(name: string, config: Config, resolver?: E
   };
 }
 
-export async function getAvatar(name: string, config: Config, resolver?: EnsResolver): Promise<string | null> {
+export async function getAvatar(name: string, config: Config, resolver?: EnsResolver): Promise<string | undefined> {
```

Comment:
Here we also will have an issue since `resolver.getText('avatar')` returns `Promise<string | null>` we return `null` and handle it later other await the promise and return `undefined`?